runtime/doc/lsp.txt

```diff
@@ -698,6 +736,66 @@ commands                                                    *vim.lsp.commands*
 
     The second argument is the `ctx` of |lsp-handler|
 
+config({name}, {cfg})                                       *vim.lsp.config()*
+    Update the configuration for an LSP client.
+
+    Use name '*' to set default configuration for all clients.
+
+    Can also be table assigned to redefine the configuration for a client.
+
+    Examples:
+    • Add a root marker for all clients: >lua
+       vim.lsp.config('*', {
+         root_markers = { '.git' },
+       })
+<
+    • Add additional capabilities to all clients: >lua
+       vim.lsp.config('*', {
+         capabilities = {
+           textDocument = {
+             semanticTokens = {
+               multilineTokenSupport = true,
+             }
+           }
+         }
+       })
+<
+    • (Re-)define the configuration for clangd: >lua
+       vim.lsp.config.clangd = {
+         cmd = {
+           'clangd',
+           '--clang-tidy',
+           '--background-index',
+           '--offset-encoding=utf-8',
+         },
+         root_markers = { '.clangd', 'compile_commands.json' },
+         filetypes = { 'c', 'cpp' },
+       }
+<
+    • Get configuration for luals: >lua
+       local cfg = vim.lsp.config.luals
+<
+
+    Parameters: ~
+      • {name}  (`string`)
+      • {cfg}   (`vim.lsp.Config`) See |vim.lsp.Config|.
+
+enable({name}, {enable})                                    *vim.lsp.enable()*
+    Enable a LSP server for filetype(s) to automatically start when opening a
+    buffer.
+
+    Uses configuration defined with `vim.lsp.config`.
+
+    Examples: >lua
+          vim.lsp.enable('clangd')
+
+          vim.lsp.enable({'luals', 'pyright'})
+<
+
+    Parameters: ~
+      • {name}    (`string|string[]`) Name(s) of client(s) to enable.
+      • {enable}  (`boolean?`) `true|nil` to enable, `false` to disable.
```

Comment:
The `|nil` is not a part of the description, not the type.

Reply:
No, it's important to have it exactly like this, since it explains what happens if `enable` is not passed.

Reply:
(Could write "`true` or `nil` to enable", if that is clearer? But then it should be done consistently since we use that pattern elsewhere (and will use even more in the future).)

Reply:
Oh I see. It looks good to me now that I read it correctly!